src/uwtools/tests/drivers/test_driver.py

```diff
@@ -3,9 +3,75 @@
 Tests for uwtools.drivers.driver module.
 """
 
+from importlib import resources
+from types import SimpleNamespace as ns
+from unittest.mock import patch
+
+import pytest
+from pytest import fixture
+
 from uwtools.drivers import driver
+from uwtools.drivers.driver import Driver
+
+
+@fixture
+def create_schema(tmp_path):
+    schema_file = tmp_path / "schema_file.jsonschema"
+    with open(schema_file, "w", encoding="utf-8") as f:
+        print("undefined_filter: '{{ 34 | not_a_filter }}'", file=f)
+    return schema_file
+
+
+@pytest.mark.usefixtures("create_schema")
+class Testinstance(Driver):
+    """
+    Test concrete class instantiation.
+    """
+
+    @property
+    def schema(self) -> str:
+        return create_schema
+
+    def requirements(self):
+        """
+        requirements.
+        """
+
+    def resources(self):
+        """
+        resources.
+        """
+
+    def output(self):
+        """
+        output.
+        """
+
+    def job_card(self):
+        """
+        job_card.
+        """
+
+
+@fixture
+def create_config(tmp_path):
+    config_file_good = tmp_path / "config_file_good.yaml"
+    config_file_bad = tmp_path / "config_file_bad.yaml"
+
+    with open(config_file_good, "w", encoding="utf-8") as f:
+        print("undefined_filter: '{{ 34 | not_a_filter }}'", file=f)
+    with open(config_file_bad, "w", encoding="utf-8") as f:
+        print("undefined_filter: '{{ 34 | not_a_filter }}'", file=f)
+
+    return str(config_file_good), str(config_file_bad)
+
+
+@pytest.mark.parametrize("vals", [("good", True), ("bad", False)])
+def test_validation(vals, create_config):
+    # test concrete classes will validate correctly
+    config_file_good, config_file_bad = create_config  # pylint: disable=unused-variable
+    cfgtype, boolval = vals
 
+    instance = Testinstance(config_file=f"config_file_{cfgtype}.yaml")
```

Comment:
I think you want to pass either `config_file_good` or `config_file_bad` to the constructor. What's being passed here is just a filename without a path, while the config files that have been created by the fixture are in some temp directory, so you have to pass those values as-is.

Reply:
```
@pytest.mark.parametrize("vals", [("good", True), ("bad", False)])
```
could be simplified to
```
@pytest.mark.parametrize("valid", [True, False])
```
and then something like
```
instance = TestDriver(config_file=config_file_good if valid else config_file_bad)
```
should work.

Reply:
Yep, and it looks better anyway - as usual, fix one bug, i++ bugs. Next it was upset at the non-jsonschema-formatted examples, so I gave them a placeholder. But now it turns out there is a peculiarity - it turns out that if you use `@pytest.mark.usefixtures` to bypass the input specification, it passes it as a function and is very stubborn to complete

Reply:
[test_driver.txt](https://github.com/ufs-community/workflow-tools/files/12247003/test_driver.txt)

Riffing on what you had, I came up with the attached. A few notes:

1. I wasn't familiar with [`usefixtures`](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#use-fixtures-in-classes-and-modules-with-usefixtures) but I think I see how that's supposed to be used. In the example, notice that `cleandir()` doesn't `return` a value -- it just does some setup work and then calls `yield`, which passes control back to the caller so that it can execute in a context/environment that is set up in some useful way. When the caller passes control back, the fixture picks up at the statement following the `yield`, where it can do whatever cleanup is needed. But the fixture doesn't supply an actual value for the test to use, which I think we need here.
2. So I simplified `TestDriver` slightly (and renamed it to `ConcreteDriver` because TIL `pylint` tries to treat `Test`-prefixed classes as containers for tests -- see [here](https://docs.pytest.org/en/latest/explanation/goodpractices.html#conventions-for-python-test-discovery)), giving it a trivial `schema()` implementation that gets patched later, in the test function.
3. I simplified the two `@fixture`s to just return strings, and let the test method create the needed temporary files. This eliminates the need to use the `tmp_path` fixture in more than one place, or to create both good and bad config files. Not a big deal, but a little more concise.
4. The JSON doc had a syntax error (a trailing comma where JSON doesn't allow one), which I discovered when running it through `jq` to get a neatly-formatted version. Fixed.
5. In the test function, I mocked the `ConcreteDriver.schema` attribute, making it a simple `str` value -- the path to the temporary schema file.

As simple as the `configs()` and `schema()` `@fixture`s are, their values probably could have just been inlined in the `test_validation()` function, but I can imagine that they might grow more complicated later, and keeping them in the separate fixtures keeps the test-function logic uncluttered. Also, these fixtures might be used by tests to come later.

Please let me know what you think.

Reply:
AH, fantastically cleaned up! Skips over the limitations of `usefixtures` and the test class - Keep It Simple, and all of that. Thank you so much for everything!